### DISCUSSION TITLE
redirect to page with params without showing the params in URL

### DISCUSSION THREAD

**Discussion by zougari47** [vercel/next.js#49813] [App Router]
previously with next/router I do like this 
```jsx

import { useRouter } from 'next/navigation'

const Button = () => {
  const router = useRouter()
  const options = {
    id: 5,
    category: 'Cat',
  }
  return (
    <div>
      <button
        onClick={() =>
          router?.push(
            {
              pathname: '/search',
              query: options,
            },
            '/search'
          )
        }
      >
        use router
      </button>
    </div>
  )
}

export default Button
```

when I try to do it with app directory I get this error (NextRouter was not mounted).
and the feature I want doesn't exist on next/navigation useRouter.

**Comment by devjiwonchoi** [comment]
Try use next/link. This is recommended when using App Router.
https://nextjs.org/docs/app/api-reference/components/link

**Reply by zougari47** [reply]
Even if I try to use the Link component I can't hide the data on the URL

**Comment by icyJoseph** [comment]
Right, so this was a way to share data between routes, which is not necessarily possible with the new router, since we've lost the `as` parameter in various places.

I think in this case, you have to roll your own context to be able to continue sharing this data. Again, this technique, while incredible useful, was just a way to get around doing "dirty" data-sharing, between pages, or really `useRouter` consumers, so we have to just own to it, and implement a proper global state manager for these. I know it is not ideal, but that's where we are, I think.

**Reply by zougari47** [reply]
The problem is I want to do data fetching on the server side and you can't access the state on the server page !!

**Reply by icyJoseph** [reply]
Right then I am not sure this is possible 😕 you'd need to give up server components for the data fetching and use the `use` hook instead, but yeah... Not ideal at all
